tools/rc/ResourceCompiler.cpp

```diff
@@ -411,36 +413,54 @@ int main(int argc, char** argv)
     }
     return_code = EXIT_FAILURE;
   }
-  
-  option::Option* bundleFileOpt = options[BUNDLEFILE];
-  if (bundleFileOpt && bundleFileOpt->count() > 1 )
+
+  // multiple specifications of the following args are illegal.
+  auto check_multiple_args = [&options, &return_code](OptionIndex optionidx, const std::string& arg)
   {
-    std::cerr << "(--bundle-file | -b) appear multiple times in the arguments. Check usage." << std::endl;
-    return_code = EXIT_FAILURE;
-  }
-  
-  option::Option* outFileOpt = options[OUTFILE];
-  if (outFileOpt && outFileOpt->count() > 1 )
+    option::Option* opt = options[optionidx];
+    if (opt && opt->count() > 1 )
+    {
+      std::cerr << arg << " appears multiple times in the arguments. Check usage." << std::endl;
+      return_code = EXIT_FAILURE;
+    }
+  };
+  check_multiple_args(BUNDLEFILE, "(--bundle-file | -b)");
+  check_multiple_args(OUTFILE, "(--out-file | -o)");
+  check_multiple_args(BUNDLENAME, "(--bundle-name | -n)");
+  check_multiple_args(MANIFESTADD, "(--manifest-add | -m)");
+
+  if (argc == 0 || options[HELP])
```

Comment:
Abhinay and I had a discussion about this. The consensus was we could change the behavior to display the usage information only if "--help" is provided. I felt that traditionally, command-line programs display only the errors. Of course, if you feel the other way, the majority wins so I'll revert back to displaying usage information for errors  :)